k2/csrc/host/CMakeLists.txt

```diff
@@ -25,7 +25,11 @@ set_target_properties(fsa PROPERTIES OUTPUT_NAME "k2fsa")
 target_link_libraries(fsa PUBLIC k2_log)
 target_link_libraries(fsa PUBLIC k2_nvtx)
 target_include_directories(fsa PUBLIC ${CUDA_TOOLKIT_INCLUDE})
-target_link_libraries(fsa PUBLIC -L${CUDA_TOOLKIT_ROOT_DIR}/targets/x86_64-linux/lib nvToolsExt)
+target_link_libraries(fsa
+  PUBLIC
+    -L${CUDA_TOOLKIT_ROOT_DIR}/targets/x86_64-linux/lib  # for /usr/local/cuda
```

Comment:
Is there a reason why you did not choose `-L${CUDA_TOOLKIT_ROOT_DIR}/lib64`? You are basically assuming an x86-64 system with your current choice.

lib64 is a symlink that is set up appropriately for you. Here's what mine looks like:

```
  /usr/local/cuda:
  total used in directory 132 available 782.7 GiB
  drwxr-xr-x 17 root root  4096 Apr 23 09:35 .
  drwxr-xr-x 13 root root  4096 Apr 23 10:10 ..
  drwxr-xr-x  3 root root  4096 Apr 23 09:35 bin
  drwxr-xr-x  5 root root  4096 Apr 23 09:35 doc
  -rw-r--r--  1 root root 60244 Apr 23 09:35 EULA.txt
  drwxr-xr-x  6 root root  4096 Apr 23 09:35 extras
  lrwxrwxrwx  1 root root    28 Apr 23 09:35 include -> targets/x86_64-linux/include
  lrwxrwxrwx  1 root root    24 Apr 23 09:35 lib64 -> targets/x86_64-linux/lib
  drwxr-xr-x  8 root root  4096 Apr 23 09:35 libnsight
  drwxr-xr-x  7 root root  4096 Apr 23 09:35 libnvvp
  drwxr-xr-x  7 root root  4096 Apr 23 09:35 nsight-compute-2019.5.0
  drwxr-xr-x  2 root root  4096 Apr 23 09:35 nsightee_plugins
  drwxr-xr-x  6 root root  4096 Apr 23 09:35 nsight-systems-2019.5.2
  drwxr-xr-x  3 root root  4096 Apr 23 09:35 nvml
  drwxr-xr-x  7 root root  4096 Apr 23 09:35 nvvm
  drwxr-xr-x  7 root root  4096 Apr 23 09:35 nvvmx
  drwxr-xr-x  3 root root  4096 Apr 23 09:35 share
  drwxr-xr-x  2 root root  4096 Apr 23 09:35 src
  drwxr-xr-x  3 root root  4096 Apr 23 09:35 targets
  drwxr-xr-x  2 root root  4096 Apr 23 09:35 tools
  -rw-r--r--  1 root root    21 Apr 23 09:35 version.txt
```

Reply:
I used
```
/usr/local/cuda$ find . -name libnvToolsExt.so
./targets/x86_64-linux/lib/libnvToolsExt.so
```
to locate `libnvToolsExt.so`.

That is the reason to choose `-L${CUDA_TOOLKIT_ROOT_DIR}/targets/x86_64-linux/lib`.

Will change to `lib64`. Thanks.